src/app/controllers/api/chat.controller.ts

```diff
@@ -0,0 +1,69 @@
+import { EventName, WebsocketContext, WebsocketResponse, WebsocketErrorResponse, ValidatePayload } from "@foal/socket.io";
+import { dependency } from "@foal/core";
+import { DB } from "../../services";
+
+export class ChatController {
+    @dependency
+    db: DB;
+
+    createRoomSchema = {
+        additionalProperties: false,
+        properties: {
+            fromUserID: { type: 'integer', minimum: 1 },
+            toUserID: { type: 'integer', minimum: 1 }
+        },
+        required: ['fromUserID', 'toUserID'],
+        type: 'object'
+    };
+    
+    sendMessageSchema = {
+        additionalProperties: false,
+        properties: {
+            fromUserID: { type: 'integer', minimum: 1 },
+            toUserID: { type: 'integer', minimum: 1 },
+            message: { type: 'string', minLength: 1 }
+        },
+        required: ['fromUserID', 'toUserID', 'message'],
+        type: 'object'
+    };
+
+    @EventName('/create-room')
```

Comment:
Mainly to have the event names follow the same format as HTTP controllers.

Reply:
I'm not sure if that's necessary, since those are defining paths rather than event names (and it's possible the http routes may be able to omit the leading `/`?), but doesn't really matter

Reply:
Discussed this separately, apparently this is a Foal convention, plus foal allows for namespacing via controllers - fine as is!